azure/scope/machinepool.go

```diff
@@ -127,6 +127,7 @@ func (m *MachinePoolScope) ScaleSetSpec() azure.ScaleSetSpec {
 		SpotVMOptions:                m.AzureMachinePool.Spec.Template.SpotVMOptions,
 		FailureDomains:               m.MachinePool.Spec.FailureDomains,
 		TerminateNotificationTimeout: m.AzureMachinePool.Spec.Template.TerminateNotificationTimeout,
+		VMExtensions:                 m.AzureMachinePool.Spec.Template.VMExtensions,
```

Comment:
why is this change needed?

Reply:
I thought that we would need to change the scale set spec since we added the VMExtensions field to the spec in case it is needed anywhere. It is currently not being used. Should I remove it if that's the case?